### ISSUE TITLE
Light/Dark mode detection no longer works on Ubuntu with Fyne 2.5

### ISSUE THREAD

**Issue by ErikKalkoken** [fyne-io/fyne#5029]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Fyne apps normally start in light or dark mode, depending on the current preferences of the OS.

Since Fyne 2.5 this does no longer seam to work on Ubuntu. I tested this with the demo app and it consistently starts in dark mode, even though the OS is set to light mode (see screenshot).

However, if I compile an app with Fyne 2.4.5 it correctly runs in light mode, so this looks to me like a new bug that was introduced with 2.5.

I also tested this on Windows 10, and there the theme detection still works with Fyne 2.5 as expected.

### How to reproduce

1. Chose "Light" style under Settings/Appearance
2. Start the Fyne demo app
3. The app runs in dark mode

### Screenshots

![Screenshot from 2024-07-25 14-35-04](https://github.com/user-attachments/assets/3b2ef216-d6e1-40e7-bfa6-ee7d0a88d48a)

### Example code

N/A

### Fyne version

2.5.0

### Go compiler version

1.22.4

### Operating system and version

Ubuntu 22.04.4 LTS

**Comment by dweymouth** [comment]
It may have been a bug introduced in moving to the `rymdport/portal` package - see difference between 

https://github.com/fyne-io/fyne/blob/release/v2.4.x/app/app_xdg.go#L37

and 

https://github.com/fyne-io/fyne/blob/release/v2.5.x/app/app_xdg.go#L39

@Jacalz any ideas?

**Comment by dweymouth** [comment]
The watchTheme code changed between v2.4 and v2.5 too:

https://github.com/fyne-io/fyne/blob/release/v2.4.x/app/app_xdg.go#L174

https://github.com/fyne-io/fyne/blob/release/v2.5.x/app/app_xdg.go#L118

@ErikKalkoken if you're able to test that switching back to the v2.4.x implementations restores the correct behavior on startup and detecting the theme switch while the app is running, we could probably get a PR merged to fix this in the meantime while waiting on issues with rymdport/portal to be resolved

**Comment by ErikKalkoken** [comment]
yes, can confirm it works correctly both on startup and while the app is running for 2.4.5 on Ubuntu. 

The live theme switching is shown in the below video with shows the same simple Fyne app, once compiled with 2.4.5 and once with 2.5:
[Screencast from 25.07.2024 18:03:28.webm](https://github.com/user-attachments/assets/591300b0-f0ca-4d29-b6fc-fd4f167770ce)

Here is the app:
```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Fyne 2.5.0")

	w.SetContent(widget.NewLabel("Hello World!"))
	w.Resize(fyne.Size{Width: 800, Height: 600})
	w.ShowAndRun()
}

```

**Comment by dweymouth** [comment]
Oh, I was asking if you would be able to test swapping out the implementations of those 2 functions with the 2.4.5 versions in the `release/v2.5.x` branch, and if that worked, if you'd consider opening a PR :)

**Comment by ErikKalkoken** [comment]
Sorry, but I know too little about that part of the Fyne code to be able fix this myself.

**Comment by dweymouth** [comment]
OK, I can probably get around to it in a day or few next time I am on my Ubuntu machine

**Comment by Jacalz** [comment]
Like mentioned in https://github.com/fyne-io/fyne/pull/5030#issuecomment-2273331907, I can not replicate this on Fedora 40. Likely not a bug within the `rymdport/portal` package as far as I can see.

**Comment by Jacalz** [comment]
I think this might be a case of the never ending problem with LTS distros using out of date versions of packages and it causing problems. In this case, Ubuntu 22.04 likely has a Gnome version that does not support the new FreeDesktop appearance interface as far as I can tell. I'd appreciate some help testing https://github.com/fyne-io/fyne/pull/5057.

**Comment by Jacalz** [comment]
I have an idea for what could be causing this. I switched from using the old deprecated `.Read` interface for the dBUS call to instead use the new `.ReadOne` call. I wonder if old Linux LTS releases like Ubuntu 22.04 perhaps don't support the new interface and thus always fails when trying to read the value?

**Comment by Jacalz** [comment]
I have opened https://github.com/fyne-io/fyne/pull/5061 now. I'm fairly certain that it should fix things once and for all. The main problem that I found on the Fyne side was that changes to how we handle dark/light mode internally had a race condition where the dark theme preference was looked up after the theme was set up. I also made sure to fall back to the `.Read` function that we used for settings in v2.4.x (hopefully fixing Ubuntu 22.04). On top of that, I improved the time ti takes to switch theme when the theme changes.
